server/users.js

```diff
@@ -596,6 +596,14 @@ class User extends Chat.MessageContext {
 		}
 		return this.group + this.name;
 	}
+	/**
+	 * @param {string} roomid
+	 */
+	getIdentityWithStatus(roomid = '') {
+		const identity = this.getIdentity(roomid);
+		if (!this.status) return identity;
+		return `${identity}@${this.isAway ? '!' : ''}${this.status}`;
```

Comment:
nit: I might make this

```ts
const isAway = this.isAway ? '!'  :  '';
return `${identity}@${isAway}${this.status};
```
but thats pretty minor and its fine as is.

Reply:
I might make the `!` part of `this.status`.

Reply:
`get isAway { return this.status.charAt(0) === '!' }`